library/build.gradle

```diff
@@ -55,6 +55,7 @@ configurations {
 
 dependencies {
     implementation 'com.android.support:support-annotations:' + rootProject.supportLibVersion
+    implementation 'com.android.support:support-compat:' + rootProject.supportLibVersion
```

Comment:
Comparing the generated library aar file without and with support-compat it looks like with support-compat it increases the library size by 709b and the contents by 4.2kb~. When it is only used for 3 simple methods and trying to keep the library as small as possible it doesn't seem worth while using. 
![image](https://user-images.githubusercontent.com/16527946/33399562-01f90abe-d54b-11e7-85ec-942791df4864.png)

Reply:
SGTM! I'll copy over the support lib implementations with a comment saying where they came from.